project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java

```
@@ -0,0 +1,138 @@
+import com.google.autograder.servlets.helpers.Services;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+@WebServlet("/createNewAssignment")
+public final class CreateNewAssignmentServlet extends HttpServlet {
+
+    private static final String END_POINT = "https://classroom.googleapis.com/v1/courses/{courseId}/courseWork";
+    private static final String HOST_URL = "https://8080-778d1d95-8447-4f8a-990b-b90da194d107.us-east1.cloudshell.dev";
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String requestURL = (String) request.getHeader("Referer");
+        int courseIDIndex = requestURL.indexOf("?courseID=") + 10;
+        String courseID = requestURL.substring(courseIDIndex);
+
+        String authorization = API.getCurrentUserAPIAuthorization();
+
+        if (authorization != null) {
+            try {
+                byte[] postBodyData = buildPostBody(request).getBytes(StandardCharsets.UTF_8);
+                HttpURLConnection connection = buildHttpURLConnection(courseID, authorization, postBodyData.length);
+                connection.getOutputStream().write(postBodyData);
+                connection.connect();
+
+                int responseCode = connection.getResponseCode();
+
+                if (responseCode == 200) {
+                
+                    System.out.println("\n\nBING\tBANG\tBONG\n\n");
+
+                } else {
+
+                    System.out.println("\n\nBAD\tRESPONSE\tCODE:\t" + responseCode + "\n\n");
+
+                }
+
+            } catch(Exception e) {
+                System.out.println(e);
+                e.printStackTrace();
+            }
+        } else {
+            System.out.println("\n" + "INVALID AUTH CODE : " + authorization);
```

This should return an error response, otherwise the redirected page will result in an error

---

In this case the OAuth flow has failed and API calls will not work. I will redirect the user to the login page.